include/nbl/builtin/hlsl/colorspace/EOTF.hlsl

```diff
@@ -0,0 +1,114 @@
+
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_COLOR_SPACE_EOTF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_COLOR_SPACE_EOTF_INCLUDED_
+
+#include <nbl/builtin/hlsl/common.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace colorspace
+{
+namespace eotf
+{
+
+float3 identity(in float3 nonlinear)
+{
+    return nonlinear;
+}
+
+float3 impl_shared_2_4(in float3 nonlinear, in float vertex)
+{
+    bool3 right = greaterThan(nonlinear, vertex.xxx);
+    return lerp(nonlinear / 12.92, pow((nonlinear + 0.055.xxx) / 1.055, (2.4).xxx), right);
```

Comment:
If we're sticking with the .xxx, there's a few ones missing here

Reply:
No, `right` and `nonlinear` are originally vectors, not a promoted scalars